File: crates/tldr-cli/tests/fixtures/vuln_migration_v1/ruby/xss_string_literal_fp.rb
# vt=Xss lang=ruby — names below are inside strings/comments only
# params[:id] is a source. ActiveRecord execute, render html_safe,
# `cmd`, File.open, Net::HTTP.get, YAML.load are sinks.

class DocsOnly
  def docs
    doc = "params[:id] flows into ActiveRecord::Base.connection.execute(SELECT ...)"
    more = "render html_safe; backtick cmd; File.open(p); Net::HTTP.get(u); YAML.load(d)"
    "#{doc} #{more}"
  end
end

File: crates/tldr-cli/tests/fixtures/vuln_migration_v1/lua/xss_string_literal_fp.lua
-- vt=Xss lang=lua — names below are inside strings/comments only
-- ngx.req.get_uri_args is a source. db:query, ngx.say, os.execute,
-- io.open are sinks. Referenced in strings only below.

local function docs()
    local doc = "ngx.req.get_uri_args flows into db:query(SELECT ... )"
    local more = "ngx.say, os.execute, io.open — string-only references here"
    return doc .. more
end
return docs
